src/main/java/org/plp/implementations/plpisa/PlpFile.java

```diff
@@ -0,0 +1,132 @@
+package org.plp.implementations.plpisa;
+
+import org.plp.isa.AsmFile;
+import org.plp.isa.AsmProgram;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is an in memory representation of an asm file {@link AsmFile}.
+ * This will map to a file path and will be part of {@link PlpProgram}
+ */
+public class PlpFile implements AsmFile {
+    private final String fileName;
+    private List<String> fileContent;
+    private final AsmProgram program;
+
+    public PlpFile(String fileName, AsmProgram program) throws IOException {
+        validateFileName(fileName, program);
```

Comment:
program.validateFileName(fileName);

Reply:
As I am going to make AsmFile independent of the AsmProgram, our only check will be if the path is not a directory.